config.m4

```diff
@@ -329,9 +332,21 @@ if test "$MONGODB" != "no"; then
 
     AC_CHECK_TYPE([socklen_t], [AC_SUBST(MONGOC_HAVE_SOCKLEN, 1)], [AC_SUBST(MONGOC_HAVE_SOCKLEN, 0)], [#include <sys/socket.h>])
 
-    AC_SUBST(MONGOC_ENABLE_COMPRESSION_SNAPPY, 0)
-    AC_SUBST(MONGOC_ENABLE_COMPRESSION_ZLIB, 0)
-    AC_SUBST(MONGOC_ENABLE_COMPRESSION, 0)
+    with_snappy=auto
+    with_zlib=auto
+    m4_include(src/libmongoc/build/autotools/m4/pkg.m4)
+    m4_include(src/libmongoc/build/autotools/CheckSnappy.m4)
+    m4_include(src/libmongoc/build/autotools/CheckZlib.m4)
```

Comment:
https://github.com/mongodb/mongo-php-driver/pull/690 originally added this down with our SASL checks, neither of which are relevant if we're using system libraries for libmongoc and libbson. Since I was touching this code anyway, I've moved this up within the condition for configuring the driver with bundled libmongoc.